src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -13,8 +13,12 @@
     /// </summary>
     internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
     {
+        /// <summary>
+        /// Conversion from stopwatch time (ticks since boot) to local time.
+        /// </summary>
+        private static readonly DateTimeOffset stopwatchEpoch = DateTime.Now - TimeSpan.FromTicks(Stopwatch.GetTimestamp());
```

Comment:
Yes, the timestamp comes from: 
https://github.com/aspnet/Hosting/blob/4abb48e1aaa730062cef7f0f0fd08c9c5b72413d/src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs#L37

For now, I'll leave it the way it is (although I will change it to use UTC to avoid time-zone ambiguity). I see the changes made to support the higher resolution clock in the shared library: https://github.com/Microsoft/ApplicationInsights-dotnet/issues/358 

Internally, there's a ```BeginTimeInTicks``` property on ```OperationTelemetry``` that ```Start()``` and ```Stop()``` set. It would be nice if those extension methods had overloads that would let me supply the timestamp in ticks, thus avoiding a second call to ```Stopwatch.GetTimestamp()```